GeocachingGroupingUICache/src/main/com/gg/uicache/incoming/JMSMessageListener.java

```diff
@@ -0,0 +1,47 @@
+package com.gg.uicache.incoming;
+
+import com.gg.generated.GeocacheByCounty;
+import com.gg.generated.GeocacheByRegion;
+import com.gg.generated.GeocacheDetail;
+import com.gg.uicache.config.HazelcastConfig;
+import com.hazelcast.core.HazelcastInstance;
+import com.hazelcast.map.IMap;
+import org.springframework.jms.annotation.JmsListener;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Resource;
+
+@Component
+public class JMSMessageListener {
+
+
+    @Resource
+    private HazelcastInstance hazelcastInstance;
+
+    @JmsListener(destination = "${jms.county.input.queue}")
+    public void countyListener(GeocacheByCounty message) {
+        if (message.getRegion() != null) {
+            IMap<String, GeocacheByCounty> map = hazelcastInstance.getMap(HazelcastConfig.COUNTY_CACHE);
+            map.put(message.getCounty(), message);
+        }
```

Comment:
Can't you have some common generic<T> function that does it for you and you can just call it from these Listener methods?